libs/algorithms/include/hpx/algorithm.hpp

```diff
@@ -0,0 +1,72 @@
+//  Copyright (c) 2020 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/parallel/algorithm.hpp>
+#include <hpx/parallel/container_algorithms.hpp>
+
+namespace hpx {
+    using hpx::parallel::adjacent_find;
+    using hpx::parallel::all_of;
+    using hpx::parallel::any_of;
+    using hpx::parallel::copy;
+    using hpx::parallel::copy_if;
+    using hpx::parallel::copy_n;
+    using hpx::parallel::count;
+    using hpx::parallel::count_if;
+    using hpx::parallel::equal;
+    using hpx::parallel::fill;
+    using hpx::parallel::fill_n;
+    using hpx::parallel::find;
+    using hpx::parallel::find_first_of;
+    using hpx::parallel::find_if;
+    using hpx::parallel::find_if_not;
+    using hpx::parallel::for_each;
+    using hpx::parallel::for_each_n;
+    using hpx::parallel::for_loop;
+    using hpx::parallel::for_loop_n;
+    using hpx::parallel::for_loop_n_strided;
+    using hpx::parallel::generate;
+    using hpx::parallel::generate_n;
+    using hpx::parallel::inplace_merge;
+    using hpx::parallel::is_heap;
+    using hpx::parallel::is_heap_until;
+    using hpx::parallel::is_partitioned;
+    using hpx::parallel::is_sorted;
+    using hpx::parallel::is_sorted_until;
+    using hpx::parallel::lexicographical_compare;
+    using hpx::parallel::merge;
+    using hpx::parallel::minmax_element;
+    using hpx::parallel::mismatch;
+    using hpx::parallel::move;
+    using hpx::parallel::none_of;
+    using hpx::parallel::partition;
+    using hpx::parallel::partition_copy;
+    using hpx::parallel::remove;
+    using hpx::parallel::remove_copy;
+    using hpx::parallel::remove_copy_if;
+    using hpx::parallel::remove_if;
+    using hpx::parallel::replace;
+    using hpx::parallel::replace_copy;
+    using hpx::parallel::replace_copy_if;
+    using hpx::parallel::replace_if;
+    using hpx::parallel::reverse;
+    using hpx::parallel::reverse_copy;
+    using hpx::parallel::rotate;
+    using hpx::parallel::rotate_copy;
+    using hpx::parallel::search;
+    using hpx::parallel::search_n;
+    using hpx::parallel::set_difference;
+    using hpx::parallel::set_intersection;
+    using hpx::parallel::set_symmetric_difference;
+    using hpx::parallel::set_union;
+    using hpx::parallel::sort;
+    using hpx::parallel::stable_partition;
+    using hpx::parallel::swap_ranges;
+    using hpx::parallel::unique;
+    using hpx::parallel::unique_copy;
+}    // namespace hpx
```

Comment:
This misses some of the algorithms (there might be more): 
- adjacent_difference 
- inner_product
- exclusive_scan 
- inclusive_scan
- find_end
- reduce
- max_element 
- min_element
- includes
- transform
- all uninitialized_xxx algorithms

However some of those belong into different headers (`hpx/numeric.hpp` and `hpx/memory.hpp`)

Reply:
Good point. I can add those headers as well.

Reply:
I'm leaving `hpx/memory.hpp` out from this PR because of #4647. I notice we have an `accumulate` in the `detail` namespace. Should we make that public or just use `std::accumulate` instead since our `accumulate` isn't (and shouldn't) be parallel anyway?

Reply:
There is no `std::accumulate(par, …)`, it's called `std::reduce`. The code in the detail namespace is just an implementation detail for our algorithms.

Reply:
I guess the question is if we can use `std::accumulate` as an implementation detail instead of the `accumulate` we have?

Reply:
Unfortunately, we cannot do that. While our accumulate is functionally equivalent to `std::accumulate` it allows for the begin and end iterators to be different, while the std version assumes that both iterators are of the same type.